.github/workflows/release.yml

```diff
@@ -0,0 +1,64 @@
+name: Release
+on:
+  release:
+    types:
+      # works for building pre-releases
+      - published
+      # works for releases and changing from pre-release to release
+      - released
+
+env: 
+ 
+  DOCKER_IMAGE: newrelic/k8s-webhook-cert-manager
+  RELEASE_VERSION: ${{ github.event.release.tag_name }}
+  S3_PATH: s3://nr-downloads-main/infrastructure_agent
+  PRERELEASE: ${{ github.event.release.prerelease }}
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Extract version from release tag
+        run: |
+          echo "RELEASE_VERSION=$(echo $RELEASE_VERSION | grep -oE '([0-9.]*[0-9])$')" >> $GITHUB_ENV
+
+      - name: Test for prerelase
+        if: ${{ github.event.release.prerelease }}
+        run: |
+          echo "RELEASE_VERSION=$RELEASE_VERSION-pre" >> $GITHUB_ENV
+          echo "S3_PATH=$S3_PATH/test" >> $GITHUB_ENV
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1 
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      - name: Build and push
+        env:
+          DOCKER_IMAGE_NAME: ${{ env.DOCKER_IMAGE }}
+          DOCKER_IMAGE_TAG: ${{ env.RELEASE_VERSION }}
+        run: |
+          make build-container
+          echo "docker push $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG"
+
+          # push 'latest' if NOT pre-release
+          if [[ -z "$PRERELEASE" || "false" == "$PRERELEASE" ]]; then
+            docker tag $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG $DOCKER_IMAGE:latest
+            echo "docker push $DOCKER_IMAGE:latest"
+          fi
+
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      
+      - name: Upload manifest to S3
+        run: |
+          aws s3 cp deploy/nri-kube-events.yaml $S3_PATH/integrations/kubernetes/k8s-webhook-cert-manager-${{ env.RELEASE_VERSION }}.yaml"
```

Comment:
Github recommends using env vars normally (i.e. `$VAR` rather than `${{ env.VAR }}` when possible. I'm not entirely sure about why, but I'd change it just for consistency with the rest of the script.

Reply:
To be honest in this case it would almost make sense to use an output value from the step instead of using env vars since it's all in a single job.